controllers/tlspolicies_validator.go

```diff
@@ -0,0 +1,96 @@
+
+func (t *ValidateTLSPoliciesValidatorReconciler) Subscription() *controller.Subscription {
+	return &controller.Subscription{
+		Events: []controller.ResourceEventMatcher{
+			{Kind: &machinery.GatewayGroupKind},
+			{Kind: &kuadrantv1alpha1.TLSPolicyKind, EventType: ptr.To(controller.CreateEvent)},
+			{Kind: &kuadrantv1alpha1.TLSPolicyKind, EventType: ptr.To(controller.UpdateEvent)},
+			{Kind: &CertManagerCertificateKind},
+			{Kind: &CertManagerIssuerKind},
+			{Kind: &CertManagerClusterIssuerKind},
+		},
+		ReconcileFunc: t.Validate,
+	}
+}
+
+func (t *ValidateTLSPoliciesValidatorReconciler) Validate(ctx context.Context, _ []controller.ResourceEvent, topology *machinery.Topology, _ error, s *sync.Map) error {
+	logger := controller.LoggerFromContext(ctx).WithName("ValidateTLSPolicyTask").WithName("Reconcile")
+
+	// Get all TLS Policies
+	policies := lo.FilterMap(topology.Policies().Items(), func(item machinery.Policy, index int) (*kuadrantv1alpha1.TLSPolicy, bool) {
+		p, ok := item.(*kuadrantv1alpha1.TLSPolicy)
+		return p, ok
+	})
+
+	// Get all gateways
+	gws := lo.FilterMap(topology.Targetables().Items(), func(item machinery.Targetable, index int) (*machinery.Gateway, bool) {
+		gw, ok := item.(*machinery.Gateway)
+		return gw, ok
+	})
+
+	isCertManagerInstalled := false
+	installed, ok := s.Load(IsCertManagerInstalledKey)
+	if ok {
+		isCertManagerInstalled = installed.(bool)
+	} else {
+		logger.V(1).Error(errors.New("isCertManagerInstalled was not found in sync map, defaulting to false"), "sync map error")
+	}
+
+	for _, policy := range policies {
+		if policy.DeletionTimestamp != nil {
+			logger.V(1).Info("tls policy is marked for deletion, skipping", "name", policy.Name, "namespace", policy.Namespace)
+			continue
+		}
+
+		if !isCertManagerInstalled {
+			s.Store(TLSPolicyAcceptedKey(policy.GetUID()), kuadrant.NewErrDependencyNotInstalled("Cert Manager"))
+			continue
+		}
+
+		// TODO: This should be only one target ref for now, but what should happen if multiple target refs is supported in the future?
+		targetRefs := policy.GetTargetRefs()
+		for _, targetRef := range targetRefs {
+			// Find gateway defined by target ref
+			_, ok := lo.Find(gws, func(item *machinery.Gateway) bool {
+				if item.GetName() == targetRef.GetName() && item.GetNamespace() == targetRef.GetNamespace() {
+					return true
+				}
+				return false
+			})
+
+			// Can't find gateway target ref
+			if !ok {
+				logger.V(1).Info("tls policy cannot find target ref", "name", policy.Name, "namespace", policy.Namespace)
+				s.Store(TLSPolicyAcceptedKey(policy.GetUID()), kuadrant.NewErrTargetNotFound(policy.Kind(), policy.GetTargetRef(), apierrors.NewNotFound(kuadrantv1alpha1.TLSPoliciesResource.GroupResource(), policy.GetName())))
+				continue
+			}
+
+			logger.V(1).Info("tls policy found target ref", "name", policy.Name, "namespace", policy.Namespace)
+			s.Store(TLSPolicyAcceptedKey(policy.GetUID()), nil)
+		}
```

Comment:
Interestingly, we may not need to find the object the policy points to.

Because policies are linked to their targets already, a policy `p` where `len(p.GetTargetRefs()) > 0 && len(topology.Targetables().Children(p)) == 0` cannot be valid.

Reply:
Obviously, what I'm suggesting won't give enough details to report in the message which target cannot be found.

Reply:
Oh cool, yeah that's a nicer way of knowing whether a policy can't find their targetRef, I'll update 👍

Reply:
`len(p.GetTargetRefs()) == len(topology.Targetables().Children(p))` is probably what we expect instead? 🤔 
I think this is fine as we currently don't support multiple target refs but this may not work if we want to report which is not found as you've said, and also not sure what the policy status should be in the case where at least one target ref in the list is not found  🤔